### DISCUSSION TITLE
How to move the messagge between box from left to right with `ortho` layout?

### DISCUSSION THREAD

**Discussion by automataIA** [plantuml-stdlib/C4-PlantUML#378] [Q&A] [answered]
How to move the message between box from left to right. Example message on the right:
![immagine](https://github.com/user-attachments/assets/1d718256-8047-4419-86b8-12ef9abcd5d0)

**Comment by Potherca** [comment]
As far as I am aware, PlantUML does not provide a `skinparam` or `<style>` element to position edge labels (i.e. arrow text messages).

If you could post the diagram (or a realistic representation of the diagram if it is confidential), we might be able to figure out a solution.

**Reply by automataIA** [reply]
this 
```
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml

!theme C4_united from https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/themes

'LAYOUT_WITH_LEGEND()
    
skinparam {
    linetype ortho
    ArrowThickness 3
    roundcorner 20
    maxMessageSize 300
    wrapWidth 300
}

Person(developer, "Developer", "GitLab User")
System_Boundary(pipeline, "GitLab CI/CD Pipeline") {
    Container(security, "Security Stage", "GitLab CI", "SAST & Dependency Scanning")
    Container(test, "Test Stage", "GitLab CI", "Unit Tests, Coverage & Linting")
    Container(build, "Build Stage", "GitLab CI", "Docker Build & Push")
    Container(deploy, "Deploy Stage", "GitLab CI", "Staging & Production Deploy")
}

System_Ext(registry, "Container Registry", "GitLab Container Registry")
System_Ext(production, "Production Environment", "Kubernetes/Cloud")
System_Ext(staging, "Staging Environment", "Kubernetes/Cloud")

Rel_Right(developer, pipeline, "Pushes code")
Rel(security, test, "Passes security checks")
Rel(test, build, "Passes tests")
Rel(build, registry, "Pushes container image")
Rel(deploy, staging, "Deploys to staging")
Rel(deploy, production, "Manual deploy to production")

SHOW_FLOATING_LEGEND()
@enduml
```

**Comment by kirchsth** [comment]
Hi @automataIA,

independent of the original problem, the !theme should be every time the first include (maybe it was only a typo in your sample, but the Rel-labels are not visible based on that)

```plantuml
@startuml
!theme C4_united from https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/themes
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
...
```

BR Helmut

[![https://www.plantuml.com/plantuml/uml/hPJ1Jzim5CRl-IjC3rih0K63qqwKjWEq0XLfXNQgNFkjiPhOqVCpq4tyxtjE4mWmjCjwYVtv-txFVhPwwaaYXR98zYY74iJeP1ci8T3Y9xfIv4IL_vgcA1yEruRoi0eUK3bBOEbGkJAj2caZuC2JBimg7PqSpAAqk9gcfVG4cDPad-mPgugWuVz0UQCZJaXZ0G-hU8BaytJuusOnNzvTpY-Mqydvv7hS7oI2VudV65j9bANuNGi5nsXRWN18kQkb8Q9xcET6RInuBuvh4LsmMZcqWEBBKIsLyl6A3N8DcVa5ulXe9p-WhEwCfhnMdf9a1kYTxMkuXy9LWFkYDsx7FPwS6vhAbLXudWwIREi9okLPh2TnswzC1N6JByRHPJeQYrcZzmRDIPwRqFUW0XhQSYHhXY8ZtcUl2ucJR9ZDnISnXWgi1glOfwIrngvxWpTG0a-ScFFd0zZ26X9ntUzpxXwGJGoV6aj_0ww2AJJ7pkBt0-JOgGsWg2sCcWMVl-Deg0gtZUPwy04fghoB246dWoBZhDWbcFYKj7sVF58VOMqyOMG-bn6tZTWblryST3dLSoNETCfExBr1Pqkm56dVmmhG0e5FHuKB-ZN4xpRUESA_uyaj5CjRiywf--ewBoas4hnGJWFRsTrvCyr5pwJtR6breN9G6z-uTvxs1Xjh55j3izHsibkoxPafvRgjtjvXUzZsCXdgMl6DzLLphwGDiX2xDSvqLcCxiekRk-MtwSrmVdbz_lA7S0fMXxBu0m00](https://www.plantuml.com/plantuml/svg/hPJ1Jzim5CRl-IjC3rih0K63qqwKjWEq0XLfXNQgNFkjiPhOqVCpq4tyxtjE4mWmjCjwYVtv-txFVhPwwaaYXR98zYY74iJeP1ci8T3Y9xfIv4IL_vgcA1yEruRoi0eUK3bBOEbGkJAj2caZuC2JBimg7PqSpAAqk9gcfVG4cDPad-mPgugWuVz0UQCZJaXZ0G-hU8BaytJuusOnNzvTpY-Mqydvv7hS7oI2VudV65j9bANuNGi5nsXRWN18kQkb8Q9xcET6RInuBuvh4LsmMZcqWEBBKIsLyl6A3N8DcVa5ulXe9p-WhEwCfhnMdf9a1kYTxMkuXy9LWFkYDsx7FPwS6vhAbLXudWwIREi9okLPh2TnswzC1N6JByRHPJeQYrcZzmRDIPwRqFUW0XhQSYHhXY8ZtcUl2ucJR9ZDnISnXWgi1glOfwIrngvxWpTG0a-ScFFd0zZ26X9ntUzpxXwGJGoV6aj_0ww2AJJ7pkBt0-JOgGsWg2sCcWMVl-Deg0gtZUPwy04fghoB246dWoBZhDWbcFYKj7sVF58VOMqyOMG-bn6tZTWblryST3dLSoNETCfExBr1Pqkm56dVmmhG0e5FHuKB-ZN4xpRUESA_uyaj5CjRiywf--ewBoas4hnGJWFRsTrvCyr5pwJtR6breN9G6z-uTvxs1Xjh55j3izHsibkoxPafvRgjtjvXUzZsCXdgMl6DzLLphwGDiX2xDSvqLcCxiekRk-MtwSrmVdbz_lA7S0fMXxBu0m00)](https://www.plantuml.com/plantuml/uml/hPJ1Jzim5CRl-IjC3rih0K63qqwKjWEq0XLfXNQgNFkjiPhOqVCpq4tyxtjE4mWmjCjwYVtv-txFVhPwwaaYXR98zYY74iJeP1ci8T3Y9xfIv4IL_vgcA1yEruRoi0eUK3bBOEbGkJAj2caZuC2JBimg7PqSpAAqk9gcfVG4cDPad-mPgugWuVz0UQCZJaXZ0G-hU8BaytJuusOnNzvTpY-Mqydvv7hS7oI2VudV65j9bANuNGi5nsXRWN18kQkb8Q9xcET6RInuBuvh4LsmMZcqWEBBKIsLyl6A3N8DcVa5ulXe9p-WhEwCfhnMdf9a1kYTxMkuXy9LWFkYDsx7FPwS6vhAbLXudWwIREi9okLPh2TnswzC1N6JByRHPJeQYrcZzmRDIPwRqFUW0XhQSYHhXY8ZtcUl2ucJR9ZDnISnXWgi1glOfwIrngvxWpTG0a-ScFFd0zZ26X9ntUzpxXwGJGoV6aj_0ww2AJJ7pkBt0-JOgGsWg2sCcWMVl-Deg0gtZUPwy04fghoB246dWoBZhDWbcFYKj7sVF58VOMqyOMG-bn6tZTWblryST3dLSoNETCfExBr1Pqkm56dVmmhG0e5FHuKB-ZN4xpRUESA_uyaj5CjRiywf--ewBoas4hnGJWFRsTrvCyr5pwJtR6breN9G6z-uTvxs1Xjh55j3izHsibkoxPafvRgjtjvXUzZsCXdgMl6DzLLphwGDiX2xDSvqLcCxiekRk-MtwSrmVdbz_lA7S0fMXxBu0m00)

**Reply by automataIA** [reply]
Thank you for the tips. But there are some command to move the message like Passes tests from left to right.

**Comment by Potherca** [comment, marked as answer]
Ah, okay, I see what the problem is.

As stated on https://plantuml.com/layout-engines, when using `skinparam linetype ortho`   the label position is wrong.

Apparently this is a known bug (see https://github.com/plantuml/backlog/issues/11)

Some details can also be found in https://forum.plantuml.net/1608

As of this writing, there is no fix for this behavior, other than not using `ortho` layout. Sorry.

**Reply by automataIA** [reply]
Unlucky
